src/main/java/de/tum/in/www1/artemis/web/rest/lecture/OnlineUnitResource.java

```diff
@@ -0,0 +1,203 @@
+
+    public OnlineUnitResource(LectureRepository lectureRepository, AuthorizationCheckService authorizationCheckService, OnlineUnitRepository onlineUnitRepository) {
+        this.lectureRepository = lectureRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.onlineUnitRepository = onlineUnitRepository;
+    }
+
+    /**
+     * GET lectures/:lectureId/online-units/:onlineUnitId: gets the online unit with the specified id
+     *
+     * @param onlineUnitId the id of the onlineUnit to retrieve
+     * @param lectureId the id of the lecture to which the unit belongs
+     * @return the ResponseEntity with status 200 (OK) and with body the online unit, or with status 404 (Not Found)
+     */
+    @GetMapping("lectures/{lectureId}/online-units/{onlineUnitId}")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<OnlineUnit> getOnlineUnit(@PathVariable Long onlineUnitId, @PathVariable Long lectureId) {
+        log.debug("REST request to get onlineUnit : {}", onlineUnitId);
+        var onlineUnit = onlineUnitRepository.findByIdElseThrow(onlineUnitId);
+        if (onlineUnit.getLecture() == null || onlineUnit.getLecture().getCourse() == null) {
+            throw new ConflictException("Lecture unit must be associated to a lecture of a course", "onlineUnit", "lectureOrCourseMissing");
+        }
+        if (!onlineUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException("Requested lecture unit is not part of the specified lecture", "onlineUnit", "lectureIdMismatch");
+        }
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, onlineUnit.getLecture().getCourse(), null);
+        return ResponseEntity.ok().body(onlineUnit);
+    }
+
+    /**
+     * PUT /lectures/:lectureId/online-units : Updates an existing online unit .
+     *
+     * @param lectureId      the id of the lecture to which the online unit belongs to update
+     * @param onlineUnit the online unit to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated onlineUnit
+     */
+    @PutMapping("/lectures/{lectureId}/online-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<OnlineUnit> updateOnlineUnit(@PathVariable Long lectureId, @RequestBody OnlineUnit onlineUnit) {
+        log.debug("REST request to update an online unit : {}", onlineUnit);
+        if (onlineUnit.getId() == null) {
+            throw new BadRequestException();
+        }
+
+        if (onlineUnit.getLecture() == null || onlineUnit.getLecture().getCourse() == null) {
+            throw new ConflictException("Lecture unit must be associated to a lecture of a course", "onlineUnit", "lectureOrCourseMissing");
+        }
+
+        // Validate the URL
+        try {
+            new URL(onlineUnit.getSource());
+        }
+        catch (MalformedURLException exception) {
+            throw new BadRequestException();
+        }
```

Comment:
Please extract those code lines into a method as we did in https://github.com/ls1intum/Artemis/pull/5187

Reply:
✅